### PR TITLE
feat: redirect dashboard group to search filter after clicking

### PR DESCRIPTION
**Pull request by feng-tao**
Signed-off-by: feng-tao <fengtao04@gmail.com>

### Summary of Changes

Once click search dashboard group, redirect to search filter page instead of external link.

### Tests

Test locally

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/pages/DashboardPage/index.tsx [hunk 1]
```diff
@@ -100,6 +103,15 @@ export class DashboardPage extends React.Component<
     }
   }
 
+  onClick = (e) => {
+    const dashboardGroup = this.props.dashboard.group_name;
```
Comment (Golodhros) [inline amundsen_application/static/js/pages/DashboardPage/index.tsx:107]: We'll usually first destructure the props, like this:
```
const { dashboard, searchDashboardGroup } = this.props;

logClick(e, {...
   label: dashboard.group_name,
});
searchDashboardGroup(dashboardGroup);

Comment (feng-tao) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: thanks @Golodhros , pr is updated

File: amundsen_application/static/js/pages/DashboardPage/index.tsx [hunk 2]
```diff
@@ -181,12 +193,12 @@ export class DashboardPage extends React.Component<
             />
             <div className="body-2">
               Dashboard in&nbsp;
+              {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
```
Comment (feng-tao) [inline amundsen_application/static/js/pages/DashboardPage/index.tsx:196]: if anyone has better suggestion to solve this, please do. I look at the link for the eslint which suggests to use a button which I don't think it matches our UX design suggestion.

Comment (Golodhros) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx:196]: Yeah, it should be a button.
Try something like this:
```
<button type="button" className="clickable-group btn btn-link" id="dashboard-group-link" onClick={this.searchGroup} >{dashboard.group_name}</button>

Comment (Golodhros) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx:196]: You wouldn't need the clickable-group class

Comment (Golodhros) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx:196]: Also, if the id is just for tracking purposes, you should use 'data-type="dashboard-group-link"' instead

Comment (feng-tao) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: cool, thanks @Golodhros . pr updated

**Review by dikshathakur3119 — APPROVED** [review]
LGTM

File: amundsen_application/static/js/pages/DashboardPage/index.tsx [hunk 3]
```diff
@@ -316,6 +327,17 @@ export class DashboardPage extends React.Component<
   }
 }
 
+export function searchDashboardGroup(
```
Comment (ttannis) [inline amundsen_application/static/js/pages/DashboardPage/index.tsx:330]: nit: don't need to export if this function is only referenced in this file.

**Review by ttannis — APPROVED** [review]

File: amundsen_application/static/js/pages/DashboardPage/index.tsx (hunk 3 above)
Comment (feng-tao) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: cool, pr updated

**Pull request merged**